pilot/pkg/serviceregistry/kube/controller/multicluster.go

```diff
@@ -318,6 +349,9 @@ func (m *Multicluster) UpdateMemberCluster(clusterID string, rc *secretcontrolle
 // when a remote cluster is deleted.  Also must clear the cache so remote resources
 // are removed.
 func (m *Multicluster) DeleteMemberCluster(clusterID string) error {
+	// Delete the cluster from the global ClusterSet.
+	m.clusterSet.DeleteCluster(model.ClusterName(clusterID))
```

Comment:
do we need it in the lock? I am not sure what the lock is needed for. I am guessing no since the clusteset handles locking, but just checking

Reply:
Yeah that's right ... ClusterSet locks.